source/real_locations.rst

```diff
@@ -0,0 +1,116 @@
+Checking the people in a location (SIR)
+=======================================
+
+What we were doing
+------------------
+
+While calculating the number of people who could potentially infect a person, we did the following:
+
+.. code-block:: scala
+
+    def computeInfectedFraction(node: Node, placeType: String, context: Context): Double = {
+    val totalNeighbourCount = node.getConnectionCount(node.getRelation[Person]().get)
+    if (totalNeighbourCount == 0)
+      return 0d
+    val infectedNeighbourCount = node.getConnectionCount(node.getRelation[Person]().get,
+      "infectionState" equ Infected).toDouble
+    infectedNeighbourCount / totalNeighbourCount.toDouble
+  }
+
+Let's take a closer look at the first line, how we calculate ``totalNeighbourCount``.
+
+.. code-block:: scala
+
+  val totalNeighbourCount = node.getConnectionCount(node.getRelation[Person]().get)
+
+Assume that the node was an ``Office``. In that case,
+
+.. code-block:: scala
+
+  node.getRelation[Person]().get
+
+returns the ``EMPLOYER_OF`` string. Now, ``totalNeighbourCount`` is counting the total number of nodes which the workplace node is the ``EMPLOYER_OF``.
+
+The problem arises when we introduce quarantining to our model. Someone who's infected may follow a different schedule, and stay at home. However, ``totalNeighbourCount`` *doesn't care* about the location a person has at a particular tick: all it does it count the number of people with the appropriate relations.
```

Comment:
The problem doesn't actually arise only when we have quarantining: it's a much more general issue, and shouldn't be linked to it. It depends very strongly on the schedule: for example, even when one _didn't_ quarantine people, suppose that an office had a "day shift" and a "night shift". In this case too there would be a problem.

As you point out, the fundamental issue is this: in counting the total number of nodes, we make _no mention_ of the individual's schedule. As a result, we are merely counting the nodes that are linked _in principle_ with this location, not those that are actually interacting with others in this location.